### PR TITLE
Add support for /nullable:warnings and /nullable:safeonlywarnings

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -63286,6 +63539,8 @@ static object M()
             assertDiagnosticOptions(NullableContextOptions.Disable);
             assertDiagnosticOptions(NullableContextOptions.Enable);
             assertDiagnosticOptions(NullableContextOptions.SafeOnly);
+            assertDiagnosticOptions(NullableContextOptions.Warnings);
+            assertDiagnosticOptions(NullableContextOptions.SafeOnlyWarnings);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Maybe I missed it, but do we have a test that shows that `safeonlywarnings` produces a subset of diagnostics produced by `warnings`?

**Comment by AlekseyTs** [comment]
@cston, @333fred Please review

**Comment by AlekseyTs** [comment]
CC @jaredpar For approval

**Comment by jaredpar** [comment]
Approved

**Pull request merged**